### PR TITLE
Update wallet adapter + misc improvements

### PR DESCRIPTION
**Pull request by jordansexton**
# Description

- Updates the versions of `@solana/wallet-adapter-*` and `@solana/web3.js` used, which include various fixes and UX improvements
- Uses the `useAnchorWallet` hook provided by `@solana/wallet-adapter-react` to handle interfacing between React and Anchor
- Adds Slope (https://slope.finance/#/wallet) since it has NFT display capability (the first 3 wallets are shown by default now)
- Adds support for the Sollet browser extension
- Adds a `.env.example` template file and adjusts the docs accordingly

## Type of change

- [x] New feature
- [x] Refactor

## How Has This Been Tested?

- [x] Developer Tools

Built and ran using `yarn start`

### CODE REVIEW

**Comment by jordansexton** [comment]
By the way if you like some of these changes and don't like others (i.e. the env changes), I can just unbundle them.

**Comment by levicook** [comment]
Hey! Just looking at this now. Thanks for the contribs and I'll get you feedback stat

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,7 @@
+REACT_APP_CANDY_MACHINE_CONFIG=__PLACEHOLDER__
+REACT_APP_CANDY_MACHINE_ID=__PLACEHOLDER__
+REACT_APP_TREASURY_ADDRESS=__PLACEHOLDER__
+REACT_APP_CANDY_START_DATE=__PLACEHOLDER__
+
+REACT_APP_SOLANA_NETWORK=devnet
+REACT_APP_SOLANA_RPC_HOST=https://explorer-api.devnet.solana.com
```
Comment (levicook) [inline .env.example:7]: thanks for adding this it was badly needed

File: .gitignore [hunk 2]
```diff
@@ -22,3 +22,5 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 .env
+.env.*
+!.env.example
```
Comment (levicook) [inline .gitignore:26]: +1

**Comment by levicook** [comment]
these changes feel totally reasonable. I see some styling / visual changes that I don't have time to check right this second. would you mind posting a screenshot? I'm willing to go off of that as documentation of visual improvements. If not, will have to circle back to view that.

**Comment by jordansexton** [comment]
>  I see some styling / visual changes that I don't have time to check right this second. would you mind posting a screenshot?

Yeah, totally. So for reference, the Material UI theme changes come verbatim from here: https://github.com/solana-labs/wallet-adapter/blob/f99da7e088cf95643cf983ebac133cb9f7c4e46a/packages/starter/material-ui-starter/src/App.tsx#L8-L34

And the CSS change comes verbatim from here: https://github.com/solana-labs/wallet-adapter/blob/f99da7e088cf95643cf983ebac133cb9f7c4e46a/packages/starter/material-ui-starter/src/index.css#L8-L9

Before:
Screen Shot 2021-09-15 at 4 54 44 PM

After:
Screen Shot 2021-09-15 at 4 54 19 PM

**Comment by jordansexton** [comment]
Users can override these however they want, but the ThemeProvider in MUI allows them to set colors and overrides that will be used throughout the app rather than at the component level. This PR just provides a sane default based on what `@solana/wallet-adapter-material-ui` uses.

**Comment by levicook** [comment]
Thank you for the added info and sorry the merge took so long!

File: .env.example (hunk 1 above)
Comment (BrixBane) [reply .env.example:7]: thanks for adding

**Pull request merged**